src/pgduckdb_background_worker.cpp

```diff
@@ -101,17 +102,20 @@ bool CanTakeBgwLockForDatabase(Oid database_oid);
 extern "C" {
 
 PGDLLEXPORT void
-pgduckdb_background_worker_main(Datum /* main_arg */) {
-	elog(LOG, "started pg_duckdb background worker");
-	if (!pgduckdb::CanTakeBgwLockForDatabase(0)) {
-		elog(LOG, "pg_duckdb background worker: could not take lock for database '%u'. Will exit.", 0);
+pgduckdb_background_worker_main(Datum main_arg) {
+	Oid database_oid = DatumGetObjectId(main_arg);
+	if (!pgduckdb::CanTakeBgwLockForDatabase(database_oid)) {
+		elog(LOG, "pg_duckdb background worker: could not take lock for database '%u'. Will exit.", database_oid);
 		return;
 	}
+
+	elog(LOG, "started pg_duckdb background worker for database %u -- MyDB: %u", database_oid, MyDatabaseId);
+
 	// Set up a signal handler for SIGTERM
 	pqsignal(SIGTERM, die);
 	BackgroundWorkerUnblockSignals();
 
-	BackgroundWorkerInitializeConnection(duckdb_motherduck_postgres_database, NULL, 0);
+	BackgroundWorkerInitializeConnectionByOid(database_oid, InvalidOid, 0);
 
 	SpinLockAcquire(&pgduckdb::BgwShmemStruct->lock);
 	pgduckdb::BgwShmemStruct->bgw_latch = MyLatch;
```

Comment:
This is problematic with multiple background workers. Only the background worker that started last will now be woken up when TriggerActivity is called.